text/0119-standard-system-metrics.md

```diff
@@ -0,0 +1,146 @@
+|-----------------------------|-----|-----------------|---------|--------------------|
+|system.filesystem.usage      |bytes|UpDownSumObserver|device   |(identifier)        |
+|                             |%    |                 |state    |used, free, reserved|
+|system.filesystem.utilization|     |UpDownSumObserver|device   |(identifier)        |
+|                             |     |                 |state    |used, free, reserved|
+
+#### `system.net.`
+
+**Description:** System level network metrics.
+|Name                       |Units|Instrument       |Label Key|Label Values                                                                                  |
+|---------------------------|-----|-----------------|---------|----------------------------------------------------------------------------------------------|
+|system.net.dropped\_packets|1    |SumObserver      |device   |(identifier)                                                                                  |
+|                           |     |                 |direction|transmit, receive                                                                             |
+|system.net.packets         |1    |SumObserver      |device   |(identifier)                                                                                  |
+|                           |     |                 |direction|transmit, receive                                                                             |
+|system.net.errors          |1    |SumObserver      |device   |(identifier)                                                                                  |
+|                           |     |                 |direction|transmit, receive                                                                             |
+|system<!--notlink-->.net.io|bytes|SumObserver      |device   |(identifier)                                                                                  |
+|                           |     |                 |direction|transmit, receive                                                                             |
+|system.net.connections     |1    |UpDownSumObserver|device   |(identifier)                                                                                  |
+|                           |     |                 |protocol |tcp, udp, [others](https://en.wikipedia.org/wiki/Transport_layer#Protocols)                   |
+|                           |     |                 |state    |[e.g. for tcp](https://en.wikipedia.org/wiki/Transmission_Control_Protocol#Protocol_operation)|
+
+#### OS Specific System Metrics - `system.{os}.`
+
+System level metrics specific to a certain operating system should be prefixed with `system.{os}.` and follow the hierarchies listed above for different entities like CPU, memory, and network. For example, metrics pertaining to Linux inodes would appear under `system.linux.filesystem.inodes.*`, reusing the `filesystem` name proposed above.
```

Comment:
Do we care about metrics that are consistent over a number of operating systems, but not all, e.g. would we have?

- `system.linux.filesystem.inodes.*`
- `system.freebsd.filesystem.inodes.*`
- `system.osx.filesystem.inodes.*`

Reply:
Something feels off about having the platform type in the metric name. Feels a little bit like when the unit type is in the metric name. Might be useful to have platform as a resource label though.

The only reason I could see for needing `system.$platform` is if there were conflicting metrics where they `system.foo` means something different on Linux than Windows but I think we're trying to avoid that situation anyway.

Reply:
I also am not sure on this. #108 mostly gives examples for runtimes like `runtime.java.*`, here's the relevant section:
> use prefixed metric names in cases where similar metrics have significantly different implementations across the breadth of all existing metrics

I agree it's not ambiguous to have `system.filesystem.inodes`, but is it valuable to have everything under the common top level `system.*` be relevant to all platforms?

Reply:
If the metrics were _truly_ platform specific it might make sense to namespace the metric. But generally I think they fall into either Windows or non-Windows, as Jame's point out. Is it useful to distinguish between system.linux.filesystem.inodes and system.freebsd.filesystem.inodes? I'm not sure that is useful.

Reply:
Just to add a counterexample where metrics of the same name can vary in implementation, Linux apparently [calculates load average differently](https://en.wikipedia.org/wiki/Load_(computing)#Unix-style_load_calculation) than many other unix systems. Since the magnitude of metric values would be different across OSes, I think a separate namespace would be called for.

Just looking at more prior art as well, Prometheus Node Exporter has a [support table](https://github.com/prometheus/node_exporter#enabled-by-default) showing a lot of metrics are not available on all unixes. These aren't necessarily ambiguous so I suppose its ok to not namespace them.

Reply:
Yeah I'm not sure what to do about load. I just feel like if most metrics start with system.{filesystem,cpu,etc.} because they're normalized but some start with system.$platform it'll be kind of weird but I hear you. I don't have a good answer.

Reply:
Is load average something that people really want to monitor? I would prefer to monitor utilization (i.e., rate of CPU seconds used per second), a value I expect to range between 0 and NumCPUs.

Reply:
@jmacd the [Collector](https://github.com/open-telemetry/opentelemetry-collector/tree/master/receiver/hostmetricsreceiver/internal/scraper/loadscraper) sends load average as does the [Prometheus Node Exporter](https://github.com/prometheus/node_exporter#enabled-by-default), but I'm not sure if that's a good justification.

@jrcamp I agree. What do you think of `$platform.system.*`?

Reply:
@aabmass what would we do about inodes where it's on pretty much every platform but windows? Would on linux and freebsd it be linux.system.filesystem.inodes and freebsd.system.filesystem.inodes?

Reply:
@jrcamp, I'm not sure. I will change this example to something unambiguous, and then capture this discussion in the "Open Questions" section.

Reply:
@aabmass are you still planning to add this discussion? Thanks.

Reply:
I moved it to "Open Questions". I don't think we've reached a conclusion still

Reply:
We agreed to merge this and move on to the specification.